KaotoIO/kaoto-next
# feat(canvas): Validate nodes upon rendering

### Context
Currently, we're validating the canvas' nodes when editing the Form, using the original model. This causes that in some cases, we get a false positive result since a parameter might have been defined using the URI field.

### Changes
This commit aims to validate each node upon rendering, meaning that the node will be validated whenever the node is rendered, which happens the first time a route is presented or when the node is selected or deselected.

In addition to that, we're parsing the node definition so we can account for the URI field when applicable.

### How to test
Using the following `YAML`, you should see that only the missing-parameters nodes are highlighted as `warning`
```yaml
- route:
    id: SqlBeansRoute
    from:
      uri: timer
      steps:
        - to:
            uri: sql
        - to:
            uri: log:test
- beans:
    - name: mysqlDataSource
      type: io.kaoto.MysqlDataSource
      properties:
        port: "3306"
        url: jdbc:mysql://localhost/test
        username: dbuser
        password: dbpass
        driver-class-name: com.mysql.jdbc.Driver
    - name: mysqlPostgreSqlSource
      type: io.kaoto.MysqlPostgreSqlSource
      properties:
        port: "5432"
        url: jdbc:postgresql://localhost/test
        username: dbuser
        password: dbpass
        driver-class-name: org.postgresql.Driver
- from:
    uri: timer:timer-1?period=5000&delay=5&synchronous=true
    steps: []
    
# camel-k: language=yaml

# Write your routes here, for example:
- from:
    uri: "jms:cheese"
    steps:
      - log: "${body}"
      
- route:
    id: route-7827
    from:
      uri: kamelet:chuck-norris-source
      steps:
        - log: ${body}
        - to:
            description: azz
            uri: kamelet:kafka-not-secured-sink?topic=foobar&bootstrapServers=localhost
            parameters: {}          
```

fix: https://github.com/KaotoIO/kaoto-next/issues/697
